src/metabase/api/table.clj

```diff
@@ -50,7 +50,7 @@
   [id :as {{:keys [display_name entity_type visibility_type description caveats points_of_interest
                    show_in_getting_started], :as body} :body}]
   {display_name            (s/maybe su/NonBlankString)
-   entity_type             (s/maybe s/Any)
+   entity_type             (s/maybe su/Any)
```

Comment:
There's no point in adding an `entity_type` validation here to be `s/Any` because that validation is effectively doing nothing. We should just remove this line entirely.

Also `(s/maybe s/Any)` is redundant because `s/Any` allows `nil` values.

Reply:
The whole point of validating API input anyway is to make sure we get valid values in. `entity_type` should be using something like `s/enum` to constrain the valid values

Reply:
I don't have the history here, but @salsakran or @sbelak might be able to chime in. I switched it recently from an `enum` to `Any` as we had a bug. I guess the `entity_type` was at one time used for something different. Now it's being used by the automagic dashboards stuff. The issue was sync added an entity type to an existing table. When the FE asked for table metadata, made an update to different metadata of that table and sent that to the backend, the old/out of date `enum` was being used to validate the data that the UI didn't do anything with.

I hadn't realized `Any` included nil. I changed it to Any as a was to document that we could get a value there but, we just passed it through. AFAIK the FE wasn't doing anything with it (showing it or changing it). I can change the schema to something else or go another route here, but I don't have much knowledge about what values are possible there.

Reply:
For example see

https://github.com/metabase/metabase/blob/master/src/metabase/util/schema.clj#L124